### PR TITLE
feat: Language server with validation

### PR DESCRIPTION
**Pull request by d-jeffery**
## References

Coupled the example language server back up, and added a validation check

image

Addresses https://github.com/openfga/language/issues/17 but still more refinement can be done.

[openfga-vscode-0.1.0.vsix.zip](https://github.com/openfga/vscode-ext/files/12261805/openfga-vscode-0.1.0.vsix.zip)

## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: server/package.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+	"name": "lsp-sample-server",
+	"description": "Example implementation of a language server in node.",
+	"version": "1.0.0",
+	"author": "Microsoft Corporation",
+	"license": "MIT",
```
Comment (d-jeffery) [inline server/package.json:6]: Will fix.

File: server/package.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
+	"name": "lsp-sample-server",
+	"description": "Example implementation of a language server in node.",
+	"version": "1.0.0",
+	"author": "Microsoft Corporation",
```
Comment (adriantam) [inline server/package.json:5]: Please update author/version/name/description etc.

File: server/src/server.ts [hunk 3]
```diff
@@ -0,0 +1,230 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
```
Comment (d-jeffery) [inline server/src/server.ts:4]: Will remove

File: server/src/server.ts [hunk 4]
```diff
@@ -0,0 +1,230 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (adriantam) [inline server/src/server.ts:2]: Should not be copyright to Microsoft

**Review by adriantam — CHANGES_REQUESTED** [review]

File: .vscode/launch.json [hunk 5]
```diff
@@ -6,23 +6,38 @@
...
+			"sourceMaps": true,
 			"args": ["--extensionDevelopmentPath=${workspaceRoot}"],
-			"outFiles": ["${workspaceRoot}/client/out/**/*.js"],
 			"preLaunchTask": "npm: webpack"
 		},
+		{
+			"type": "node",
+			"request": "attach",
+			"name": "Attach to Server",
+			"port": 6012,
```
Comment (adriantam) [inline .vscode/launch.json:19]: maybe add comments on why 6012?

**Review by adriantam — APPROVED** [review]

**Pull request merged**